src/checkin/services/flow/step/ValidateVisaStep.ts

```diff
@@ -0,0 +1,51 @@
+import { Builder } from 'builder-pattern'
+import { Context } from '../../../model/Context'
+import { Session } from '../../../model/session'
+import { FlightData } from '../../../model/session/FlightData'
+import StepTemplate from '../StepTemplate'
+import { UserInformation } from '../../../model/session/UserInformation'
+import { RequestData } from '@http/Request'
+
+export default class ValidateVisaStep extends StepTemplate {
+  private readonly COUNTRY_REQUIRE_VISA = 'US'
+  private readonly VISA_WAIVER_COUNTRIES: string[] = ['AD', 'AU', 'AT', 'BE', 'BN', 'CL', 'CZ', 'DK', 'EE', 'FI', 'FR', 'DE', 'GR', 'HU', 'IS', 'IE', 'IT', 'JP', 'LV', 'LI', 'LT', 'LU', 'MT', 'MC', 'NL', 'NZ', 'NO', 'PL', 'PT', 'SM', 'SG', 'SK', 'SI', 'KR', 'ES', 'SE', 'CH', 'TW', 'GB', 'US']
+
+  onExecute(context: Context): Promise<boolean> {
+    const session: Session = context.getSession()
+    const requestData = context.getRequest()
+    if (requestData.fields?.visa_number == null) {
+      context = requestVisaFieldInResponse(context)
+      return Promise.resolve(false)
+    }
+    context = saveVisaInSession(context, session, requestData)
+    return Promise.resolve(true)
+  }
+
+  when(context: Context): boolean {
+    const session: Session = context.getSession()
+    const flights: FlightData[] = session.data.flights
+    const isFliyingToUS: boolean = flights.some((flight) => flight.to.country === this.COUNTRY_REQUIRE_VISA)
```

Comment:
Typos in isFliyingToUS, should be isFlyingToUS. Also if we add more countries to COUNTRY_REQUIRE_VISA then I don't think isFlyingToUS would be the best name.